source/adapters/level_zero/program.cpp

```diff
@@ -180,11 +192,25 @@ UR_APIEXPORT ur_result_t UR_APICALL urProgramBuild(
   }
 
   // We no longer need the IL / native code.
-  Program->Code.reset();
-  Program->ZeModule = ZeModule;
+  hProgram->Code.reset();
+  hProgram->ZeModule = ZeModule;
   return Result;
 }
 
+UR_APIEXPORT ur_result_t UR_APICALL urProgramCompileExp(
+    ur_program_handle_t
+        hProgram,        ///< [in][out] handle of the program to compile.
+    uint32_t numDevices, ///< [in] number of devices
+    ur_device_handle_t *phDevices, ///< [in][range(0, numDevices)] pointer to
+                                   ///< array of device handles
+    const char *pOptions           ///< [in][optional] pointer to build options
+                                   ///< null-terminated string.
+) {
+  std::ignore = numDevices;
+  std::ignore = phDevices;
+  return urProgramCompile(nullptr, hProgram, pOptions);
```

Comment:
Does `urProgramCompile` actually work with a `nullptr` context? 

We do still have access to the context here:

```suggestion
  return urProgramCompile(hProgram->Context, hProgram, pOptions);
```

I'm a bit dubiour about this implementation though. I thought the issue these new interfaces are supposed to solve is to compile for one device at a time rather than all devices in a context. `urProgramCompile()` is defined as compiling for all devices in the given context, rather than *only*. the ones in the `phDevices` list.

Reply:
Hello @kbenzie ,

I fixed the urProgramCompileExp to use the context handle from the program handle for level zero. It does not use that handle so it was not necessary, but it does not hurt to pass it thru.

The urProgramCompile() for Level Zero does not do anything device specific for Level Zero so the changes for the urProgramCompileExp function did not directly impact the Level Zero adapter, but other adapters would need that functionality if urProgramCompileExp were implemented for those Adapters.

Reply:
Okay, that makes sense. Thanks!